### PR TITLE
FEAT: URL parametrisation for direction to specific algorithms and their parameter inputs; Partial implementation of Share function.

### PR DESCRIPTION
**Pull request by MaryHuang8**
URL Format Example: 

With parameter input: 
http://localhost:3000/?alg=heapSort&mode=sort&list=1,2,3,41

Without parameter input: 
http://localhost:3000/?alg=heapSort&mode=sort

Checkout ```docs/URL-Parametrisation-Requirements.pdf``` for more parameter types.

NOTE: For Graph algorithms, the URL parameters 'min' and 'max' are backdoors and are not visible on the user interface.

### CODE REVIEW

**Comment by MinniMinMin** [comment]
@Sraccity Ready for review

**Comment by Sraccity** [comment]
issues : Merge conflict needs resolving
@MinniMinMin

File: .idea/workspace.xml
Comment (Sraccity) [inline .idea/workspace.xml:1]: ## **nitpick**: add .idea into gitignore

So that workspaces/IDE changes will not be stored here

**Comment by MaryHuang8** [comment]
Thank you. I will close this pull request and publish a new version with the latest version that integrates with the Share feature.

**Pull request closed without merging**